src/internal/Subscriber.ts

```diff
@@ -113,6 +113,39 @@ export class Subscriber<T> extends Subscription implements Observer<T> {
     }
   }
 
+  /**
+   * @ignore
+   * @internal
+   * Synchronous errors thrown within `subscribe` and caught by `_trySubscribe`
+   * can only be reported using a subscriber's `error` method if the subscriber
+   * is not stopped. This method traverses the `destination` chain of
+   * subscribers and uses the appropriate unhandled mechanism to report the
+   * error if a stopped subscriber is found.
+   */
+  reportError(err?: any): void {
+    let observer: PartialObserver<T> = this;
+    while (observer) {
+      if (observer instanceof Subscriber) {
```

Comment:
I'm not sure about this check. When would `this` not be the Subscriber? If for some reason it gets wrapped in a `SafeSubscriber`?

Reply:
`this` will always be a `Subscriber`. It's just that `observer` is the iteration variable. `PartialObserver<T>` is the type of the `destination` property - which is assigned to `observer` during the iteration.

The iteration starts with `this` so that its `isStopped` property is checked. And it's all done in the one loop so that the `hostReportError` call, etc. is just in the one place.